### PR TITLE
chore: Update demo-rest to use the new Cerbos Go SDK

### PR DESCRIPTION
**Pull request by SuperAayush**
Updated the Go SDK to use the new Cerbos Go SDK

I think the current Cerbos Go SDK uses `1.20` go version and we should also change the below to match the go versions:
https://github.com/cerbos/demo-rest/blob/25244fac36fc4f50c2e8fe3c0f7013fc16ae91be/go.mod#L3

Signed-off-by: Aayush Sharma [aayusharma1401@gmail.com](mailto:aayusharma1401@gmail.com)

cc - @charithe

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,36 +3,37 @@ module github.com/cerbos/demo-rest
 go 1.18
 
 require (
-	github.com/cerbos/cerbos v0.17.0
+	github.com/cerbos/cerbos-sdk-go/cerbos v0.2.3
```
Comment (charithe) [inline go.mod:6]:
```suggestion
	github.com/cerbos/cerbos-sdk-go v0.2.3
```

**Review by charithe — COMMENTED** [review]
Yes, the Go version should be updated to 1.22 because everything up to 1.20 is EOL now. There are some compilation errors in this PR. We no longer use the `Client` interface and use the concrete type `*cerbos.GRPCClient` instead.

File: service/service.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func toInventoryResource(i db.InventoryRecord) *cerbos.Resource {
 
 // Service implements the store API.
 type Service struct {
-	cerbos    cerbos.Client
+	cerbos    cerbos.GRPCClient
```
Comment (charithe) [inline service/service.go:55]:
```suggestion
	cerbos    *cerbos.GRPCClient
```

**Review by charithe — COMMENTED** [review]
You need to run `go mod tidy` and also please sign all your commits.

```
git rebase HEAD~3 --signoff
```

File: service/service.go [hunk 3]
```diff
@@ -12,8 +12,8 @@ import (
 	"net/http"
 	"strconv"
 
-	"github.com/cerbos/cerbos/client"
-	cerbos "github.com/cerbos/cerbos/client"
+	"github.com/cerbos/cerbos-sdk-go/cerbos"
+	client "github.com/cerbos/cerbos-sdk-go/cerbos"
```
Comment (charithe) [inline service/service.go:16]: Why is the same package imported twice?

Comment (SuperAayush) [reply service/service.go:16]: If I don't import `client` as: `client "github.com/cerbos/cerbos-sdk-go/cerbos"`

It throws me the below error in `principalContext` function:
undefined: client
Line Number:
https://github.com/SuperAayush/demo-rest/blob/25244fac36fc4f50c2e8fe3c0f7013fc16ae91be/service/service.go#L159

Comment (SuperAayush) [reply service/service.go:16]: Taking reference from here for solving the above mentioned error: https://github.com/cerbos/cerbos-sdk-go/blob/aa488281f36f177ca3215c95b6b26949569c0add/README.md?plain=1#L41

Comment (charithe) [reply service/service.go:16]: Instead of importing the package twice you need to change the package references in code.

Comment (SuperAayush) [reply service/service.go]: Made the changes please review.

**Pull request merged**